torch/distributed/algorithms/model_averaging/utils.py

```diff
@@ -19,18 +19,45 @@ def average_parameters(
     if dist._rank_not_in_group(group_to_use):
         return
 
+    if isinstance(params, types.GeneratorType):
```

Comment:
For this case, actually we can also filter out any parameter where its gradient is `None`, right?

Reply:
yes, your are right.
model.parameters() input and param groups input can deal with a unified process, including gradient filter.
i will try to make it elegant.

Reply:
this filter param process has been moved to PeriodicModelAverager() and HierarchicalModelAverager()
please cr again